### PR TITLE
Improved document guide with a schema and extended information

### PR DESCRIPTION
**Pull request by bidoubiwa**
Since this is the core place where users will understand how we structure data it must be very well explained to minimize friction.

If the user does not understand the primary key while having a primary key that is not automatically inferred, his experience in MeiliSearch will not be seamless.

### CODE REVIEW

**Comment by curquiza** [comment]
Top moumoute this schema 👌

**Comment by bidoubiwa** [comment]
Do you think it is useful to add the value? Also I changed the direction of the arrows based on Thomas's suggestion.


Yes: ❤
Doc structure

No: 🎉
Doc structure (1)

File: guides/main_concepts/documents.md [hunk 1]
```diff
@@ -86,9 +81,11 @@ If none has been found, no documents will be added.
 
 ❗️ If you get the `document id error`, it means MeiliSearch could not recognize your primary key. This means your primary key does not answer the formatting explained above. To solve this error, you could send the [primary key's name as a query parameter](/references/documents.md#add-or-replace-documents) or [update your index to add the primary key's name](/references/indexes.md#create-an-index) as explained in [setting the primary key](/guides/main_concepts/documents.md#primary-key).
 
-### Primary key value format
+### Document Id
+
+The document id is the value associated to the primary key.
 
-The primary key **value** may contain only `A-Z a-z 0-9` and `-_` characters.
+The document id may contain only `A-Z a-z 0-9` and `-_` characters.
```
Comment (matboivin) [inline guides/main_concepts/documents.md]:
```suggestion
The document id must contain only `A-Z a-z 0-9` and `-_` characters.
```

File: guides/main_concepts/documents.md [hunk 2]
```diff
@@ -3,20 +3,15 @@
...
+![document structure](/document_structure.svg)
 
 #### Wording
 
-- **Attribute**: `"id"`, `"title"`, `"description"` and `"type"`.
-- **Field**: the combination of attributes and data (i.e, `"title": "Interstellar"`)
-- **Document**: The combination of all the fields between brackets.
+- **Attribute**: The name of a field.
+- **[Field](/guides/main_concepts/documents.md#fields)**: the combination of attributes and data.
+- **Document**: The collection of all the fields between brackets.
```
Comment (matboivin) [inline guides/main_concepts/documents.md]:
```suggestion
- **Document**: A document is an object which collects a list of fields in curly brackets.
```

File: guides/main_concepts/documents.md [hunk 3]
```diff
@@ -3,20 +3,15 @@
...
-}
-```
+![document structure](/document_structure.svg)
 
 #### Wording
 
-- **Attribute**: `"id"`, `"title"`, `"description"` and `"type"`.
-- **Field**: the combination of attributes and data (i.e, `"title": "Interstellar"`)
-- **Document**: The combination of all the fields between brackets.
+- **Attribute**: The name of a field.
```
Comment (matboivin) [inline guides/main_concepts/documents.md]:
```suggestion
- **Attribute**: An attribute is the name of a field, like a key.
```

File: guides/main_concepts/documents.md (hunk 2 above)
Comment (bidoubiwa) [reply guides/main_concepts/documents.md]:
```suggestion
- **Document**: A document is an object which collects a list of fields between curly brackets.
```

File: guides/main_concepts/documents.md [hunk 4]
```diff
@@ -3,20 +3,15 @@
...
+- **[Field](/guides/main_concepts/documents.md#fields)**: the combination of attributes and data.
```
Comment (matboivin) [inline guides/main_concepts/documents.md]:
```suggestion
- **[Field](/guides/main_concepts/documents.md#fields)**: A field, or a key-value pair, is a set of two data items linked together: an attribute and its corresponding data
```

File: guides/main_concepts/documents.md (hunk 2 above)
Comment (matboivin) [reply guides/main_concepts/documents.md]: Yes looks good!

**Pull request merged**